connectors/src/admin/cli.ts

```diff
@@ -99,6 +103,63 @@ const connectors = async (command: string, args: parseArgs.ParsedArgs) => {
   }
 };
 
+const github = async (command: string, args: parseArgs.ParsedArgs) => {
+  switch (command) {
+    case "test-repo": {
+      if (!args.wId) {
+        throw new Error("Missing --wId argument");
+      }
+      if (!args.dataSourceName) {
+        throw new Error("Missing --dataSourceName argument");
+      }
+      if (!args.owner) {
+        throw new Error("Missing --owner argument");
+      }
+      if (!args.repo) {
+        throw new Error("Missing --repo argument");
+      }
+
+      const connector = await Connector.findOne({
+        where: {
+          type: "github",
+          workspaceId: args.wId,
+          dataSourceName: args.dataSourceName,
+        },
+      });
+
+      if (!connector) {
+        throw new Error(
+          `Could not find connector for workspace ${args.wId}, data source ${args.dataSourceName}`
+        );
+      }
+
+      const installationId = connector.connectionId;
+      const { tempDir, files, directories } = await processRepository(
+        installationId,
+        args.owner,
+        args.repo,
+        "999"
+      );
+
+      files.forEach((f) => {
+        console.log(f);
+      });
+      directories.forEach((d) => {
+        console.log(d);
+      });
+
+      console.log(
+        `Found ${files.length} files in ${directories.length} directories`
+      );
+      console.log(
+        `Files total size: ${files.reduce((acc, f) => acc + f.sizeBytes, 0)}`
+      );
+
+      await cleanUpProcessRepository(tempDir);
```

Comment:
Should we wrap the code above with a `try/catch/finally` so we always remove it afterward?

Reply:
This is just for testing and the only thing that can really fail is processRepository, but if it fails we don't know the tempDir :/

I'll add a clean-up in a try/catch/finally in processRepository